packages/SwingSet/src/kernel/deviceTranslator.js

```diff
@@ -81,14 +97,67 @@ export function makeDSTranslator(deviceID, deviceName, kernelKeeper) {
     return ks;
   }
 
+  function translateVatstoreGet(key) {
+    insistValidVatstoreKey(key);
+    kdebug(`syscall[${deviceID}].vatstoreGet(${key})`);
+    return harden(['vatstoreGet', deviceID, key]);
+  }
+
+  function translateVatstoreSet(key, value) {
+    insistValidVatstoreKey(key);
+    assert.typeof(value, 'string');
+    kdebug(`syscall[${deviceID}].vatstoreSet(${key},${value})`);
+    return harden(['vatstoreSet', deviceID, key, value]);
+  }
+
+  function translateVatstoreGetAfter(priorKey, lowerBound, upperBound) {
+    if (priorKey !== '') {
+      insistValidVatstoreKey(priorKey);
+    }
+    insistValidVatstoreKey(lowerBound);
+    if (upperBound) {
+      insistValidVatstoreKey(upperBound);
+    }
+    kdebug(
+      `syscall[${deviceID}].vatstoreGetAfter(${priorKey}, ${lowerBound}, ${upperBound})`,
+    );
+    return harden([
+      'vatstoreGetAfter',
+      deviceID,
+      priorKey,
+      lowerBound,
+      upperBound,
+    ]);
+  }
+
+  function translateVatstoreDelete(key) {
+    insistValidVatstoreKey(key);
+    kdebug(`syscall[${deviceID}].vatstoreDelete(${key})`);
+    return harden(['vatstoreDelete', deviceID, key]);
+  }
```

Comment:
Yeah, maybe. For now, devices don't get the same syscalls as vats: they don't do promises, so `syscall.send` isn't quite right, they can't `syscall.exit`, they don't participate in GC so no `syscall.dropImports`.

But in the long run, yeah, we want devices to be more like vats, and at that point it'll make sense to drop `deviceTranslator.js` and use `vatTranslator` in its place.

Reply:
I was thinking more along the lines of a syscall translator that's parameterized according to whether it thinks it's doing the job for a vat or a device, which would allow differences between the two to be accommodated but would allow the 95% that's the same to be a single piece of code to maintain.